proxy/deploy.go

```diff
@@ -36,21 +37,26 @@ func DeployFunction(fprocess string, gateway string, functionName string, image
 		fprocessTemplate = "python3 index.py"
 	}
 
+	if (registryAuth != "") && !strings.HasPrefix(gateway, "https") {
+		fmt.Println("WARNING! Communication is not secure, please consider using HTTPS. Letsencrypt.org offers free SSL/TLS certificates.")
+	}
+
 	gateway = strings.TrimRight(gateway, "/")
 
 	if replace {
 		DeleteFunction(gateway, functionName)
 	}
 
 	req := requests.CreateFunctionRequest{
-		EnvProcess:  fprocessTemplate,
-		Image:       image,
-		Network:     network,
-		Service:     functionName,
-		EnvVars:     envVars,
-		Constraints: constraints,
-		Secrets:     secrets, // TODO: allow registry auth to be specified or read from local Docker credentials store
-		Labels:      &labels,
+		EnvProcess:   fprocessTemplate,
+		Image:        image,
+		RegistryAuth: registryAuth,
```

Comment:
I don't think that we need to store credentials in the function like this, for security reason. We might implement a method on gateway so that the gateway can register credentials to docker on the gateway.